src/Parser/Warrior/Arms/Modules/Talents/Rend.js

```diff
@@ -0,0 +1,72 @@
+import React from 'react';
+
+import Wrapper from 'common/Wrapper';
+import SPELLS from 'common/SPELLS';
+import SpellLink from 'common/SpellLink';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import Enemies from 'Parser/Core/Modules/Enemies';
+import Analyzer from 'Parser/Core/Analyzer';
+
+const EXECUTE_RANGE = 0.2;
+
+class RendAnalyzer extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+    enemies: Enemies,
+  };
+
+  rendsInExecuteRange = 0;
+  targets = [];
+  failedTargets = [];
+
+  on_initialized() {
+		this.active = this.combatants.selected.hasTalent(SPELLS.REND_TALENT.id);
+	}
+
+  on_byPlayer_cast(event) {
+    if(SPELLS.REND_TALENT.id === event.ability.guid) {
+      // If Rend is used flag the target to check their health on the next damage event.
+      this.targets.push(event.targetID);
+    }
+  }
+
+  on_byPlayer_damage(event) {
+    const targetId = event.targetID;
+
+    if(SPELLS.REND_TALENT.id === event.ability.guid && this.targets.indexOf(targetId) !== -1) {
+      // Check if the target's health was within execute range.
+      if(event.hitPoints / event.maxHitPoints < EXECUTE_RANGE) {
```

Comment:
It's prefered to avoid nested if statements by checking for the opposite and returning early instead. Applies to both of the above if statements. (you can move `the this.targets.splice(this.targets.indexOf(targetId), 1);` line up to right after the first if statement passes since it is removed no matter what at that point)